server/src/main/kotlin/fi/oph/kitu/csvparsing/CsvParser.kt

```diff
@@ -84,36 +100,41 @@ class CsvParser(
     /**
      * Converts retrieved String response into a list that is the type of Body.
      */
-    inline fun <reified T> convertCsvToData(csvString: String): List<TypedResult<T, CsvExportError>> {
-        if (csvString.isBlank()) {
-            event.add("serialization.isEmptyList" to true)
-            return emptyList()
-        }
-
-        event.add("serialization.isEmptyList" to false)
-
-        val csvMapper = getCsvMapper<T>()
-        val schema = getSchema<T>(csvMapper)
-        val lineSeparator =
-            onlyOrNull(schema.lineSeparator)
-                // This error would be probably due to internal changes so we don't pass it to normal error handling.
-                ?: throw IllegalStateException(
-                    "Can't find only one line seperator from schema (${schema.lineSeparator}).",
-                )
-
-        return csvMapper
-            .readerFor(T::class.java)
-            .with(schema)
-            .readValues<T?>(csvString)
-            .toTypedResults { index, e ->
-                val context = runCatching { csvString.split(lineSeparator)[index] }.getOrNull()
-
-                when (e) {
-                    is InvalidFormatException -> InvalidFormatCsvExportError(index, context, e)
-                    else -> SimpleCsvExportError(index, context, e)
+    final inline fun <reified T> convertCsvToData(csvString: String): List<TypedResult<T, CsvExportError>> =
+        tracer
+            .spanBuilder("CsvParser.convertCsvToData")
+            .startSpan()
+            .use { span ->
+
+                if (csvString.isBlank()) {
+                    span.setAttribute("serialization.isEmptyList", true)
```

Comment:
Parempi pattern voisi olla iffin ulkopuolella sanoa  `span.setAttribute("serialization.isEmptyList", csvString.isBlank())`. Tällöin ei tarvitse kahta kertaa samaa attribuuttia asettaa eri koodipoluissa.